src/guidescanpy/flask/core/genome.py

```diff
@@ -342,6 +344,33 @@ def query(
             if max_gc is not None:
                 results = results[results["gc-content"] <= max_gc]
 
+            if pattern_avoid is not None:
+                wildcard_to_nuc = {"N": "ACTG", "V": "ACG"}
+                patterns_avoid = [pattern_avoid]
+
+                for wildcard in wildcard_to_nuc:
+                    patterns_avoid.extend(
+                        [
+                            pattern.replace(wildcard, x)
+                            for x in wildcard_to_nuc[wildcard]
+                            for pattern in patterns_avoid
+                            if wildcard in pattern
+                        ]
+                    )
+
+                patterns_avoid = [
+                    pattern
+                    for pattern in patterns_avoid
+                    if ("N" not in pattern and "V" not in pattern)
+                ]
+
+                exclude_reg = (
```

Comment:
I think since you're using `contains` for the DataFrame, you can simplify your regular expression to remove the `^` and `$` anchors (and also the wildcards).